fdbclient/DatabaseConfiguration.cpp

```diff
@@ -198,7 +199,9 @@ bool DatabaseConfiguration::isValid() const {
 	      (usableRegions == 1 || regions.size() == 2) && (regions.size() == 0 || regions[0].priority >= 0) &&
 	      (regions.size() == 0 || tLogPolicy->info() != "dcid^2 x zoneid^2 x 1") &&
 	      // We cannot specify regions with three_datacenter replication
-	      (perpetualStorageWiggleSpeed == 0 || perpetualStorageWiggleSpeed == 1))) {
+	      (perpetualStorageWiggleSpeed == 0 || perpetualStorageWiggleSpeed == 1) &&
+	      (perpetualStorageWiggleLocality.find(':') != std::string::npos ||
+	       !perpetualStorageWiggleLocality.compare("0")))) {
```

Comment:
nit: I would write this as `perpetualStorageWiggleLocality == "0"`. Yes, that technically does an unnecessary allocation*, but the performance here isn't critical. I had to look up the `compare` API to validate this because I forgot its semantics, but `==` is pretty straightforward.

\* The allocation could be alleviated with a static/conversion to `std::string_view`, but that seems like unnecessary complexity.

Reply:
Done.